cirq/ops/linear_combinations_test.py

```diff
@@ -562,3 +562,73 @@ def test_linear_combination_of_operations_has_correct_pauli_expansion(
 ))
 def test_operation_expressions(expression, expected_result):
     assert_linear_combinations_are_equal(expression, expected_result)
+
+
+def test_pauli_sum_construction():
+    q = cirq.LineQubit.range(2)
+    pstr1 = cirq.X(q[0]) * cirq.X(q[1])
+    pstr2 = cirq.Y(q[0]) * cirq.Y(q[1])
+    psum = pstr1 + pstr2
+    assert list(psum) == [pstr1, pstr2]
+
+    psum2 = cirq.PauliSum.from_pauli_strings([pstr1, pstr2])
+    assert psum == psum2
+
+def test_pauli_sub():
+    q = cirq.LineQubit.range(2)
+    pstr1 = cirq.X(q[0]) * cirq.X(q[1])
+    pstr2 = cirq.Y(q[0]) * cirq.Y(q[1])
+    psum = pstr1-pstr2
+
+    psum2 = cirq.PauliSum.from_pauli_strings([pstr1, -1*pstr2])
+    assert psum == psum2
+
+def test_pauli_sub_simplify():
+    q = cirq.LineQubit.range(2)
+    pstr1 = cirq.X(q[0]) * cirq.X(q[1])
+    pstr2 = cirq.X(q[0]) * cirq.X(q[1])
+    psum = pstr1-pstr2
+
+    psum2 = cirq.PauliSum.from_pauli_strings([])
+    assert psum == psum2
+
+def test_paulisum_validation():
+    q = cirq.LineQubit.range(2)
+    pstr1 = cirq.X(q[0]) * cirq.X(q[1])
+    pstr2 = cirq.Y(q[0]) * cirq.Y(q[1])
+    with pytest.raises(ValueError) as e:
+        cirq.PauliSum([pstr1, pstr2])
+    assert e.match("Consider using")
+
+
+def test_add_number_paulisum():
+    q = cirq.LineQubit.range(2)
+    pstr1 = cirq.X(q[0]) * cirq.X(q[1])
+    psum = cirq.PauliSum.from_pauli_strings([pstr1]) + 1.3
+    assert psum == cirq.PauliSum.from_pauli_strings([pstr1,
+                                                     cirq.PauliString({}, 1.3)])
+
+
+def test_add_number_paulistring():
+    q = cirq.LineQubit.range(2)
+    pstr1 = cirq.X(q[0]) * cirq.X(q[1])
+    psum = pstr1 + 1.3
+    assert psum == cirq.PauliSum.from_pauli_strings([pstr1,
+                                                     cirq.PauliString({}, 1.3)])
+
```

Comment:
Add a `__repr__` test using `cirq.testing.assert_equivalent_repr`.

Reply:
nifty! and done